File: nagasenacs/src/Nagasena/Proc/IO/ByteAlignedScanner.cs
using System.IO;

using AlignmentType = Nagasena.Proc.Common.AlignmentType;

namespace Nagasena.Proc.IO {

  internal sealed class ByteAlignedScanner : SimpleScanner {

    /*
     * Constructor is intentionally non-public.
     * Use ScannerFactory to instantiate a ByteAlignedScanner.
     */
    internal ByteAlignedScanner() : base(false) {
    }

    public override AlignmentType AlignmentType {
      get {
        return AlignmentType.byteAligned;
      }
    }

    protected internal override bool readBoolean(Stream istream) {
      // use m_inputStream irrespective of istream
      return ByteAlignedCommons.readBoolean(m_inputStream);
    }

    protected internal override int readNBitUnsigned(int width, Stream istream) {
      // use m_inputStream irrespective of istream
      if (width != 0) {
        return ByteAlignedCommons.readNBitUnsigned(width, m_inputStream);
      }
      return 0;
    }

    protected internal override int readEightBitsUnsigned(Stream istream) {
      return ByteAlignedCommons.readEightBitsUnsigned(m_inputStream);
    }
  }

}

File: fujitsu/src/org/openexi/fujitsu/proc/io/ByteAlignedScanner.java
package org.openexi.fujitsu.proc.io;

import java.io.IOException;
import java.io.InputStream;

import org.openexi.fujitsu.proc.common.AlignmentType;

public final class ByteAlignedScanner extends SimpleScanner {

  /*
   * Constructor is intentionally non-public.
   * Use ScannerFactory to instantiate a ByteAlignedScanner.
   */
  ByteAlignedScanner() {
    super(false);
  }
  
  @Override
  public final AlignmentType getAlignmentType() {
    return AlignmentType.byteAligned;
  }

  @Override
  protected boolean readBoolean(InputStream istream) throws IOException {
    // use m_inputStream irrespective of istream
    return ByteAlignedCommons.readBoolean(m_inputStream);
  }

  @Override
  public int readNBitUnsigned(int width, InputStream istream) throws IOException {
    // use m_inputStream irrespective of istream
    return ByteAlignedCommons.readNBitUnsigned(width, m_inputStream);
  }
  
}
